components/api/src/main/java/com/hotels/styx/api/plugins/spi/PluginMeterRegistry.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    public RequiredSearch get(String name) {
+        return registry.get(name);
+    }
+
+    public Counter counter(String name, Iterable<Tag> tags) {
+        Iterable<Tag> newTags = injectPluginNameTag(tags);
+        return registry.counter(name, newTags);
+    }
+
+    public Counter counter(String name, String... tags) {
+        String[] newTags = injectPluginNameTag(tags);
+        return registry.counter(name, newTags);
+    }
+
+    public DistributionSummary summary(String name, Iterable<Tag> tags) {
+        Iterable<Tag> newTags = injectPluginNameTag(tags);
+        return registry.summary(name, newTags);
+    }
+
+    public DistributionSummary summary(String name, String... tags) {
+        String[] newTags = injectPluginNameTag(tags);
+        return registry.summary(name, newTags);
+    }
+
+    public Timer timer(String name, Iterable<Tag> tags) {
+        Iterable<Tag> newTags = injectPluginNameTag(tags);
+        return registry.timer(name, newTags);
+    }
+
+    public Timer timer(String name, String... tags) {
+        String[] newTags = injectPluginNameTag(tags);
+        return registry.timer(name, newTags);
+    }
+
+    public MeterRegistry.More more() {
+        return registry.more();
+    }
+
+    @Nullable
+    public <T> T gauge(String name, Iterable<Tag> tags, T stateObject, ToDoubleFunction<T> valueFunction) {
+        Iterable<Tag> newTags = injectPluginNameTag(tags);
+        return registry.gauge(name, newTags, stateObject, valueFunction);
+    }
+
+    @Nullable
+    public <T extends Number> T gauge(String name, Iterable<Tag> tags, T number) {
+        Iterable<Tag> newTags = injectPluginNameTag(tags);
+        return registry.gauge(name, newTags, number);
+    }
+
+    @Nullable
+    public <T extends Number> T gauge(String name, T number) {
+        return registry.gauge(name, Tags.of(pluginNameTag), number);
+    }
+
+    @Nullable
+    public <T> T gauge(String name, T stateObject, ToDoubleFunction<T> valueFunction) {
+        return registry.gauge(name, Tags.of(pluginNameTag), stateObject, valueFunction);
+    }
+
+    @Nullable
+    public <T extends Collection<?>> T gaugeCollectionSize(String name, Iterable<Tag> tags, T collection) {
+        Iterable<Tag> newTags = injectPluginNameTag(tags);
+        return registry.gaugeCollectionSize(name, newTags, collection);
+    }
+
+    @Nullable
+    public <T extends Map<?, ?>> T gaugeMapSize(String name, Iterable<Tag> tags, T map) {
+        Iterable<Tag> newTags = injectPluginNameTag(tags);
+        return registry.gaugeMapSize(name, newTags, map);
+    }
+
+    @Incubating(since = "1.1.0")
+    @Nullable
+    public Meter remove(Meter meter) {
+        return registry.remove(meter);
+    }
+
+    @Incubating(since = "1.1.0")
+    @Nullable
+    public Meter remove(Meter.Id mappedId) {
+        return registry.remove(mappedId);
+    }
+
+    @Incubating(since = "1.2.0")
+    public void clear() {
+        registry.clear();
```

Comment:
This will clear out every plugin's meters, and styx core's. Either limit it to this plugin's meters, or remove this method.

Reply:
Updated 👍